pkg/controller/timeseriesprediction/time_series_prediction_controller.go

```diff
@@ -54,18 +59,32 @@ func (tc *Controller) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Res
 		if errors.IsNotFound(err) {
 			if last, ok := tc.tsPredictionMap.Load(req.NamespacedName.String()); ok {
 				if tsp, ok := last.(*predictionapi.TimeSeriesPrediction); ok {
-					tc.removeTimeSeriesPrediction(tsp)
+					tc.stopPrediction(tsp)
 					return ctrl.Result{}, nil
 				}
 				return ctrl.Result{}, fmt.Errorf("assert tsp failed for tsp %v in map", req.String())
 			}
-			klog.V(4).Infof("Failed to load exist tsp %v", req.String())
-			return ctrl.Result{}, nil
+			klog.V(4).Infof("Failed to load cached tsp %v, maybe the craned is reboot", req.String())
+			return ctrl.Result{}, client.IgnoreNotFound(err)
 		}
 		klog.V(4).Infof("Failed to get TimeSeriesPrediction %v err: %v", klog.KObj(p), err)
 		return ctrl.Result{Requeue: true}, err
 	}
 
+	if !p.ObjectMeta.GetDeletionTimestamp().IsZero() {
```

Comment:
`if p.DeletionTimestamp != nil`

Reply:
> `if p.DeletionTimestamp != nil`

it is ok when the time is nil, the IsZero will judge nil pointer and return true